internal/resources/ekscluster/data_source.go

```diff
@@ -157,6 +160,8 @@ func setResourceData(d *schema.ResourceData, eksCluster *eksmodel.VmwareTanzuMan
 
 		if pos, ok := npPosMap[np.FullName.Name]; ok {
 			nodepools[pos] = npDef
+			// Add tf file nodepool tags as part of nodepool tags so that it will not show difference
+			nodepools[pos].Spec.Tags = npDataMap[np.FullName.Name].Spec.Tags
```

Comment:
Why are we suppressing the diff here? Don't we want to know if the tags on the TMC side have drifted? Won't this hide any legitimate difference in the tags between Terraform's state and TMC also?

Reply:
Since we are copying cluster level tags to nodepool tags at time of creation. So when we are applying same tf file without any changes second time, it  is showing deleted as part of tf diff changes. So, to not show those changes, we are just copying nodepool tags  which are present in tf file.
If you have any other suggestion for a better way of handling. It would be more helpful.

Reply:
Done

Reply:
@asutoshpalai ,  when cluster will have same tag with same key then same key and value will be not allowed in nodepool because cluster tags are getting populated to nodepool directly. So, logic to filter out cluster tags from nodepool will have to ignore same key and value.